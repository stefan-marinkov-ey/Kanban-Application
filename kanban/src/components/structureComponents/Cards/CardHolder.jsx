import React from "react";
import { threeDots } from "../../../utility/constantsText";
import { makeLighterColor } from "../../../utility/utilityFunc";

const CardHolder = ({ cardName, cardColor }) => {
  const elementP = makeLighterColor(cardColor);

  return (
    <div className="cardFunctionality">
      <p onClick={console.log("Placeholder For Editing Card")} style={elementP}>
        {cardName} <span>{threeDots}</span>
      </p>
    </div>
  );
};

export default CardHolder;
