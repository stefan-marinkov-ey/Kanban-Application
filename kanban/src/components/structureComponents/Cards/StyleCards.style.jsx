const { default: styled } = require("styled-components");

export const StyleCards = styled.div`
  width: 90%;
  margin: 0 auto;
  margin-bottom: 5%;
  .holder {
    display: flex;
    justify-content: space-between;
    align-items: center;
    button {
      width: 10%;
      height: 2rem;
      background: steelblue;
      color: whitesmoke;
      margin-bottom: auto;
    }
  }
`;
