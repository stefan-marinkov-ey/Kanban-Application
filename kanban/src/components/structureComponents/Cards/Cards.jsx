import React from "react";
import { deleteX } from "../../../utility/constantsText";
import Button from "../../reusableComponents/Button/Button";
import { CardDivStyled } from "../../styleComponents/Container/CardStyle";
import CardHolder from "../CardHolder/CardHolder";
import useModal from "../Modal/useModal";
import Modal from "../Modal/Modal";
import DeleteCard from "../DeleteCard/DeleteCard";

const Cards = ({
  cardName,
  cardColor,
  cardId,
  refreshCard,
  setRefreshCard,
  cardDesc,
}) => {
  const { isShowing, toggle } = useModal();
  return (
    <CardDivStyled>
      <div>
        <CardHolder
          cardName={cardName}
          cardColor={cardColor}
          cardId={cardId}
          cardDesc={cardDesc}
          refreshCard={refreshCard}
          setRefreshCard={setRefreshCard}
        />
        <Button label={deleteX} onClick={toggle} />

        <Modal isShowing={isShowing} hide={toggle}>
          <DeleteCard
            cardName={cardName}
            cardId={cardId}
            refreshCard={refreshCard}
            setRefreshCard={setRefreshCard}
            toggle={toggle}
          />
        </Modal>
      </div>
    </CardDivStyled>
  );
};

export default Cards;
