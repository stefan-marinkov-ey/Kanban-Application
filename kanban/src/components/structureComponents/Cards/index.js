import Cards from "./Cards";

export default Cards;
export { Cards };
