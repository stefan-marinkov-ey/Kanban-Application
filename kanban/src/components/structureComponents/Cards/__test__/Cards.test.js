import React from "react";
import { shallow } from "enzyme";
import Cards from "../Cards";

describe("Card", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Cards />);
  });
  it("render component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
