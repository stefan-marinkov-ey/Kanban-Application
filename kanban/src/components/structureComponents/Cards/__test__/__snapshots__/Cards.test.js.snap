// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render component 1`] = `ShallowWrapper {}`;

exports[`Holder render component 1`] = `ShallowWrapper {}`;
